### ISSUE TITLE
BibsAndAttachedToSolr export not generating shelflist item manifests correctly

### ISSUE THREAD

**Issue by jthomale** [unt-libraries/catalog-api#16]
When running a `BibsAndAttachedToSolr` compound export process, the `final_callback` method on the exporter for each individual record type (including both bibs and items) runs after all chunks have been processed. When the `shelflist.export.ItemsToSolr.final_callback` method runs, it attempts to create shelflist item manifests and store them in Redis--i.e., for each location in the record set, it sorts all of the items in Solr at that location by normalized call number and assigns row numbers.

While working on a separate issue and running a `BibsAndAttachedToSolr` export process on a single bib record, I noticed that the relevant shelflist item manifest was not being generated: when I looked at the `shelflist_item` resources at the location that should have had the manifest created, the `rowNumber` field was null after running the export.

The Celery log did log a "Creating shelflist item manifests" message, which indicates that the `shelflist.export.ItemsToSolr.final_callback` method **did** run. And, running one of the item records at that location through the `ItemsToSolr` process did correctly generate the shelflist item manifest.

I suspect that the problem may be [here](https://github.com/unt-libraries/catalog-api/blob/481c3c6f51366cb2c8ec9e9395f03e54ba28a05b/django/sierra/shelflist/exporters.py#L34). When run as an `ItemsToSolr` job by itself, `exp.get_records()` returns the set of item records that were just loaded, and pulling out the set of locations associated with those items gives you the correct set of locations to generate shelflist item manifests for. But, when run as part of a `BibsAndAttachedToSolr` job, I think `exp.get_records()` is probably returning the set of bib records associated with that job, which means the following lines aren't pulling out item location codes correctly.

**Comment by jthomale** [comment]
I'd add, although I assigned this a "priority high" label, the affect this is having on our production data should be negligible. The `BibsAndAttachedToSolr` process does run once a night to sync bib records from Sierra, but the `ItemsToSolr` process runs by itself during the day every two hours and correctly regenerates shelflist manifests each time it runs.

It could theoretically cause a problem if the call number on a bib record were changed and neither the attached items nor any of the items at that location were updated the next day. In that case, the worst you'd have would be the items attached to that bib record would appear to be out of order in the shelflist (and therefore in the inventory app) until an item at that location were run through ItemsToSolr. As far as I know, we haven't yet run into that after 9 months of using the inventory app.
